File: java-example/src/test/java/org/onflow/examples/java/executeScript/ExecuteScriptAccessAPIConnectorTest.java
package org.onflow.examples.java.executeScript;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.onflow.flow.common.test.FlowEmulatorProjectTest;
import org.onflow.flow.common.test.FlowTestClient;
import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.FlowScriptResponse;

import java.util.List;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
class ExecuteScriptAccessAPIConnectorTest {

    @FlowTestClient
    private FlowAccessApi accessAPI;

    private ExecuteScriptAccessAPIConnector scriptExecutionExample;

    @BeforeEach
    void setup() {
        scriptExecutionExample = new ExecuteScriptAccessAPIConnector(accessAPI);
    }

    @Test
    void canExecuteSimpleScript() {
        FlowScriptResponse result = scriptExecutionExample.executeSimpleScript();

        assertNotNull(result, "Result should not be null");
        assertEquals(15, result.getJsonCadence().decodeToAny());
    }

    @Test
    void canExecuteComplexScript() {
        FlowScriptResponse result = scriptExecutionExample.executeComplexScript();

        assertNotNull(result, "Result should not be null");

        // Decode the result as a list of maps
        List<Map<String, Object>> rawList = (List<Map<String, Object>>) result.getJsonCadence().decodeToAny();

        assertNotNull(rawList, "Decoded list should not be null");
        assertEquals(1, rawList.size(), "Expected exactly one StorageInfo object");

        // Manually map the LinkedHashMap to your StorageInfo class
        Map<String, Object> rawMap = rawList.get(0);
        ExecuteScriptAccessAPIConnector.StorageInfo storageInfo = new ExecuteScriptAccessAPIConnector.StorageInfo(
                (int) rawMap.get("capacity"),
                (int) rawMap.get("used"),
                (int) rawMap.get("available")
        );

        // Verify the fields of the StorageInfo object
        assertEquals(1, storageInfo.getCapacity(), "Expected capacity to be 1");
        assertEquals(2, storageInfo.getUsed(), "Expected used to be 2");
        assertEquals(3, storageInfo.getAvailable(), "Expected available to be 3");
    }
}

File: kotlin-example/src/test/kotlin/org/onflow/examples/kotlin/executeScript/ExecuteScriptAccessAPIConnectorTest.kt
package org.onflow.examples.kotlin.executeScript

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.onflow.flow.common.test.FlowEmulatorProjectTest
import org.onflow.flow.common.test.FlowTestClient
import org.onflow.flow.sdk.*

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
class ScriptExecutionExampleTest {

    @FlowTestClient
    lateinit var accessAPI: FlowAccessApi

    private lateinit var scriptExecutionExample: ExecuteScriptAccessAPIConnector

    @BeforeEach
    fun setup() {
        scriptExecutionExample = ExecuteScriptAccessAPIConnector(accessAPI)
    }

    @Test
    fun `Can execute simple script`() {
        val result = scriptExecutionExample.executeSimpleScript()

        assertNotNull(result, "Result should not be null")
        assertEquals(15, result.jsonCadence.decode<Int>())
    }

    @Test
    fun `Can execute complex script`() {
        val result = scriptExecutionExample.executeComplexScript()

        val storageInfoList = result.jsonCadence.decode<List<ExecuteScriptAccessAPIConnector.StorageInfo>>()

        assertNotNull(storageInfoList, "Storage info list should not be null")
        assertEquals(1, storageInfoList.size, "Expected exactly one StorageInfo object")

        val storageInfo = storageInfoList[0]

        assertEquals(1, storageInfo.capacity, "Expected capacity to be 1")
        assertEquals(2, storageInfo.used, "Expected used to be 2")
        assertEquals(3, storageInfo.available, "Expected available to be 3")
    }
}
